### PR TITLE
Upgrade okhttp to 4.10.0

### PR DESCRIPTION
**Pull request by wendigo**
Release notes are not needed.

### CODE REVIEW

**Review by hashhar — APPROVED** [review]
The update itself looks good to me however I'll defer the merge to @electrum on this.

**Comment by sajjoseph** [comment]
Thanks for this change. I have been using OkHttp 4.10.0 in production with Trino for a while.
You could address the TODO entry in JsonResponse.java with OkHttp upgrade. 

[https://github.com/trinodb/trino/blob/19c4275e6c78d4e773d63424ed9b1d40140655bb/client/trino-client/src/main/java/io/trino/client/JsonResponse.java#L116](https://github.com/trinodb/trino/blob/19c4275e6c78d4e773d63424ed9b1d40140655bb/client/trino-client/src/main/java/io/trino/client/JsonResponse.java#L116)

**Comment by wendigo** [comment]
@sajjoseph does it work? :)

**Comment by sajjoseph** [comment]
> @sajjoseph does it work? :)
It does! 

Matter of fact I added HTTP/2 support in our version of Trino (I am contributing that change to airlift shortly) and OkHttp module automatically switched to HTTP/2 to communicate with Coordinator. I haven't seen any issues in our test environment.

We will be promoting it to prod in a week and so we will see how older and newer clients (JDBC, CLI etc with different OkHttp versions) are behaving.

**Comment by wendigo** [comment]
That's great news @sajjoseph

**Comment by wendigo** [comment]
Ping @electrum

File: client/trino-client/src/main/java/okhttp3/internal/tls/LegacyHostnameVerifier.java [hunk 1]
```diff
@@ -13,20 +13,29 @@
  */
```
Comment (kokosing) [inline client/trino-client/src/main/java/okhttp3/internal/tls/LegacyHostnameVerifier.java:13]: > Remove OkHostnameVerifier usage

Did you mean inline `OkHostnameVerifier`? If so then please update the commit message. Also please mention `Util.verifyAsIpAddress`

**Review by kokosing — APPROVED** [review]

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**